# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by scripts/Create-CopyPatch.
# 
# T2 SDE: architecture/powerpc64/package/*/0700-ps3_gelic-dma_mask.patch
# Copyright (C) 2021 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

--- linux-5.12/drivers/net/ethernet/toshiba/ps3_gelic_net.c.vanilla	2021-05-12 19:05:18.527874803 +0200
+++ linux-5.12/drivers/net/ethernet/toshiba/ps3_gelic_net.c	2021-05-12 19:09:44.503864198 +0200
@@ -1643,6 +1643,7 @@
 	struct net_device *netdev;
 	u64 lpar_id, laid, hwconfig, eurus_lpar_access, junk;
 	int result;
+	static u64 dummy_mask = DMA_BIT_MASK(32);
 
 	pr_debug("%s: called\n", __func__);
 
@@ -1742,6 +1742,7 @@
 	gelic_card_get_vlan_info(card);
 
 	card->link_mode = GELIC_LV1_ETHER_AUTO_NEG;
+	dev->core.dma_mask = &dummy_mask;
 
 	/* setup interrupt */
 	result = lv1_net_set_interrupt_status_indicator(bus_id(card),
